src/schedule/ll_schedule.c

```diff
@@ -331,9 +331,11 @@ static void schedule_ll_domain_clear(struct ll_schedule_data *sch,
 	if (count == 1) {
 		sch->domain->registered[cpu_get_id()] = false;
 
-		count = atomic_sub(&sch->domain->num_clients, 1);
-		if (count == 1)
-			domain_clear(sch->domain);
+		if (atomic_read(&sch->domain->num_clients)) {
+			count = atomic_sub(&sch->domain->num_clients, 1);
```

Comment:
What happens if num_clients changes here between reading and subtracting. @lgirdwood @plbossart what do you think ?

Reply:
> What happens if num_clients changes here between reading and subtracting. @lgirdwood @plbossart what do you think ?

We hold the domain->lock during these checks so that should not happen.

BTW, we are kind of mess in this client/task management, I am working on refining them, please see the PR here: https://github.com/thesofproject/sof/pull/4089

Reply:
ok, this is fine with the locks held.